runtime/appruntime/metrics/metrics.go

```diff
@@ -0,0 +1,50 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+
+	"encore.dev/beta/errs"
+)
+
+type Manager struct {
+	exp Exporter
+}
+
+type Exporter interface {
+	IncCounter(name string, tags ...string)
+	Observe(name string, key string, value float64, tags ...string)
+}
+
+func NewManager(exp Exporter) *Manager {
+	return &Manager{exp: exp}
+}
+
+func (m *Manager) ReqEnd(service, endpoint string, err error, httpStatus int, durSecs float64) {
+	code := code(err, httpStatus)
+	m.exp.IncCounter(
+		"e_requests_total",
+		"service", service,
+		"endpoint", endpoint,
+		"code", code,
+	)
+	m.exp.Observe(
+		"e_request_duration_seconds",
+		"duration", durSecs,
+		"service", service,
+		"endpoint", endpoint,
+		"code", code,
+	)
+}
+
+func code(err error, httpStatus int) string {
+	if err != nil {
+		e := errs.Convert(err).(*errs.Error)
+		return e.Code.String()
+	}
+
+	if httpStatus == 0 || httpStatus == http.StatusOK {
+		return errs.OK.String()
+	}
+	return "http_" + strconv.Itoa(httpStatus)
```

Comment:
I think if the status code is one of the ones we recognize we should probably output that instead. You can use `errs.HTTPStatusToCode(httpStatus)` to get that, and fall back to `http_N` if that function reports `errs.Unknown`

Reply:
I think that the reason why it's implemented this way is to cover the case where an error happens (so `err` isn't `nil`), but `httpStatus` code is `0`. As far as I understand, this is possible?